sigal/gallery.py

```diff
@@ -785,3 +796,21 @@ def worker(args):
         return process_file(args)
     except KeyboardInterrupt:
         pass
+
+
+def album_worker(args):
+    path, f, settings = args
+    ext = splitext(f)[1]
+    media = None
+    if ext.lower() in settings['img_extensions']:
+        media = Image(f, path, settings)
+    elif ext.lower() in settings['video_extensions']:
+        media = Video(f, path, settings)
+
+    # The logger is not pickable, so it cannot be used in another process.
+    # But it is a cached property, so let's delete it
+    try:
+        del media.logger
+    except AttributeError:
+        pass
+    return media
```

Comment:
This should be done in the Media class, with the `__setstate__` and `__getstate__` methods: (dropping the logger in `__getstate__` and recreating it in `__setstate__`):
https://docs.python.org/3/library/pickle.html#pickle-state